pyuvdata/uvflag/uvflag.py

```diff
@@ -683,6 +634,140 @@ def __init__(
                 "list, tuple, string, pathlib.Path, UVData, or UVCal."
             )
 
+    def _set_telescope_requirements(self):
```

Comment:
I think the real issue comes down to `x_orientation` as we discussed on the telecon. We can't really require it for UVData because many existing file formats do not have it or require it and there's no good way to automatically fill it in. But we do require it on UVCal. Unless we decide to back off on that requirement, we're stuck with these differences. I did make the things that are always required on all objects always be required on Telescope and removed those from these methods to make it clearer what the differences are.

Reply:
One probably not very satisfying way of solving this would be to require `x_orientation` everywhere, and force people to pass it into the `read()` method for file formats that don't include it.

Reply:
Okay, I think I'm content to leave it be for now -- `x_orientation` is probably where this runs aground, but I realized that I had some ideas on how to retire this (as part of addressing #854), and maybe this could be just left as is for now (though `instrument` could probably be made to be always optional -- actually not sure why it's required for `UVData` at the moment).